apm-agent-core/src/main/java/co/elastic/apm/agent/logging/AgentMDC.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.logging;
+
+import co.elastic.apm.agent.sdk.state.GlobalState;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Agent internal MDC for log-correlation to be used when instrumented library does not provide one.
+ */
+@GlobalState
+public class AgentMDC {
```

Comment:
[for reviewer] this is a duplication of the `JulMdc` class within `ecs-logging-java`, duplicating allows to instrument `JulMdc` to use this class when agent is present, otherwise dealing with two distinct copies of `JulMdc` in two different classloaders (one in the app, one in the agent for ecs reformatting) is more complex and less clear.